main.go

```diff
@@ -0,0 +1,175 @@
+// and empty string with corresponding error if something is wrong
+func parseURL(link string) (string, error) {
+	_, err := url.ParseRequestURI(link)
+	if err != nil {
+		return "", err
+	}
+	resp, err := http.Get(link)
+	if err != nil {
+		return "", fmt.Errorf("wget: error connecting url :%v", err)
+	}
+	defer resp.Body.Close()
+
+	bodySize, err := strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 0)
+	if err != nil {
+		return "", fmt.Errorf("wget: content length = %v :%v", bodySize, err)
+	}
+	return link, nil
+}
+
+func main() {
+
+	var linksList []string
+	// parsing urls
+	// if correct, adding them to slice
+	// else print error
+	for _, inputLink := range os.Args[1:] {
+		parsedLink, err := parseURL(inputLink)
+		if err == nil {
+			linksList = append(linksList, parsedLink)
+		} else {
+			fmt.Fprintf(os.Stderr, "wget: error parsing url %v: %v\n", inputLink, err)
+		}
+	}
+
+	var fileList []*File
+	// creating WaitGroup for download goroutines
+	wg := &sync.WaitGroup{}
+
+	for _, link := range linksList {
+		file := &File{}
+		fileList = append(fileList, file)
+		wg.Add(1)
+		go file.Download(link, wg)
+	}
+
+	// we use channel to wait until the last
+	// Print function call will be terminated correctly
+	printFinishedCh := make(chan struct{})
+	// creating context variable in order to stop Print func
+	// exactly in time we need it to be stopped
+	ctx, cancel := context.WithCancel(context.Background())
+	go Print(ctx, fileList, printFinishedCh)
+
+	// waiting for all download goroutines
+	// to finish their work
+	wg.Wait()
+	// cancel ctx context which causes print func
+	// to return
+	cancel()
+	// ensure that last print will finish correctly
+	<-printFinishedCh
+	fmt.Println()
+}
+
+// Print iterates over given slice of Files, computes
+// percentages of download of all Files and shows them in stdout
+// in pseudo-table format until ctx.Done() call
+func Print(ctx context.Context, fileList []*File, printFinishedCh chan struct{}) {
+	defer close(printFinishedCh)
+
+	print := func() {
+		fmt.Print("\r")
+		for _, file := range fileList {
+			if file.totalSize != 0 {
+				percentage := float64(file.counter) / float64(file.totalSize) * 100
+				fmt.Printf("%.2f %% ", percentage)
+			} else {
+				fmt.Print("0.00 %% ")
+			}
+		}
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			print()
+			return
+		default:
+			print()
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
+}
+
+// Download tries to download file from remote address using
+// given url. Takes WaitGroup pointer to provide parallelism.
+// Prints error to os.Stderr and return if something goes wrong.
+func (file *File) Download(link string, wg *sync.WaitGroup) {
+	defer wg.Done()
+	resp, err := http.Get(link)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "wget: error connecting url :%v\n", err)
+		return
+	}
+	defer resp.Body.Close()
+
+	bodySize, err := strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 0)
```

Comment:
so, you already getting and checking content-length in parse url? why you need to do double work?